crates/biome_formatter/src/printer/mod.rs

```diff
@@ -768,7 +811,11 @@ enum Indention {
     Level(u16),
 
     /// Indent the content by n-`level`s using the indention sequence specified by the printer options and `align` spaces.
-    Align { level: u16, align: NonZeroU8 },
+    Align {
```

Comment:
Wonder if it is possible to change the `Indention` to 
```rust
enum Indention {
  Level,
  Align(NonZeroU8)
}
```
then the indent stack would be like `[Indent, align(1), Indent]`. 

Consequently, the `indent` in `PrintElementArgs` 
https://github.com/biomejs/biome/blob/fbab39f57d5a3544c0f2b99918575ec14bf17f89/crates/biome_formatter/src/printer/call_stack.rs#L29
would become redundant because we can calculate indentation based on the indent stack. What do you think?

Reply:
Thank you so much for the review.
I also worried about redundant at first.
I might be wrong, but when I try to eliminate duplication, I realized that every time we need to calculate the indentation, we have to iterate through the entire indent stack to calculate it. I don't think it's a huge performance issue, but that's why I just left it as is.

I'm still not sure which method is the best.
@Sec-ant 
If you have any thoughts, please let me know.

Reply:
That depends on what is more important to you: the performance gain or easy maintenance.

Reply:
@mdm317 Sorry I didn't have time to catch up with the conversation. I think @ah-yu has a better understanding on this than I do. If you worry about the code performance I think you can request a review from the core contributors like @ematipico, @Conaclos.

Reply:
Performance is at stake here, and I believe the indentation calculation could be considered a hot path.

Maintenance friction could be partially compensated with good internal documentation.

We could use both approaches and benchmark both of them. 

We have some benchmarks around formatting in `xtask/bench`.

Reply:
Thank you! It appears there is an obvious performance degradation for large files.

> To eliminate redundancy, how about moving the methods that calculate indentation, which are currently implemented in the PrintElementArgs struct, to the IndentStack trait, without changing the structure of indentation?

I think it is a good way to go.

Reply:
While moving the indentation logic, I found that if the current element is a LineSuffix, the current indention is saved and used later when flush_line_suffixes is called. Therefore, I implemented the SuffixStack trait to save the current indentation at this [commit](https://github.com/biomejs/biome/pull/2134/commits/d70c7c9a37621924d375b5655e361dc955786c46).